### PR TITLE
resolve some logic error within the matching algorithm

### PR DESCRIPTION
**Pull request by Nanxi-Huang**
resolve error when parsing frontend request
resolve error when some of the criteria for matching is none 
enable cancelMatch to end a matching session and an ongoing matched session

### CODE REVIEW

**Review by Linda124 — DISMISSED** [review]
Looks good to me!

File: peer-prep/src/matching-service/controllers/matchingController.js [hunk 1]
```diff
@@ -2,16 +2,30 @@ const matchingService = require('../services/matchingService');
...
+                res.status(200).json(matchResult);
+                break;
+            case 'error': 
+                res.status(500).json(matchResult);
+                break;
+            case 'cancel':
+                res.status(200).json(matchResult);
+                break;
+            default:
+                res.status(500).json({message:'Unknown error. Please try again!'});
```
Comment (laihuiqi) [inline peer-prep/src/matching-service/controllers/matchingController.js:23]: Good handle of various types of response! 👍 👍

File: peer-prep/src/matching-service/services/matchingService.js [hunk 2]
```diff
@@ -25,22 +28,23 @@ async function findMatch(request) {
...
             console.log('Successfully connected to RabbitMQ');
 
-            const criteria = `${request.language}
-                            .${request.proficiency}
-                            .${request.difficulty}
-                            .${request.topic}`;
+            const criteria = `${request.language || 'None'}
+                            .${request.proficiency || 'None'}
+                            .${request.difficulty || 'None'}
+                            .${request.topic || 'None'}`;
```
Comment (laihuiqi) [inline peer-prep/src/matching-service/services/matchingService.js:34]: Good edge cases consideration! 😄

File: peer-prep/src/matching-service/services/matchingService.js [hunk 3]
```diff
@@ -107,6 +115,18 @@ async function findMatch(request) {
...
+    for (let field of fields) {
+        if (requestCriteria[field] !== "None" &&
+            currentCriteria[field] !== "None" &&
+            requestCriteria[field] !== currentCriteria[field]) {
+            return false;
+        }
+    }
+    return true;
+}
+
```
Comment (laihuiqi) [inline peer-prep/src/matching-service/services/matchingService.js:129]: Good extraction of criteria check. The code is more neat now 😄

**Review by laihuiqi — APPROVED** [review]
Overall good job in considering edge cases and different response types. 😄🎉

**Review by Linda124 — APPROVED** [review]

**Pull request merged**